### PR TITLE
Added implementation of Lister

### PR DESCRIPTION
**Pull request by romaninsh**
Implementation of Lister see #11, #23 and #27. Simple use:

``` php
$app->add(['Lister', 'defaultTemplate'=>'lister.html'])->setSource([
    ['icon'=>'map marker', 'title'=>'Krolewskie Jadlo', 'descr'=>'An excellent polish restaurant, quick delivery and hearty, filling meals'],
    ['icon'=>'map marker', 'title'=>'Xian Famous Foods', 'descr'=>'A taste of Shaanxi\'s delicious culinary traditions, with delights like spicy cold noodles and lamb burgers.'],
    ['icon'=>'check', 'title'=>'Sapporo Haru', 'descr'=>'Greenpoint\'s best choice for quick and delicious sushi'],
]);
```

screen shot 2017-12-21 at 04 18 07

or

``` php
$view = $app->add(['View', 'template' => new \atk4\ui\Template('
<div class="ui header">Top 20 countries (alphabetically)</div>
{List}<div class="ui icon label"><i class="{iso}ae{/} flag"></i> {name}andorra{/}</div>{/}
</div>')]);

$view->add('Lister', 'List')
    ->addHook('beforeRow', function($l){ $l->current_row['iso']=strtolower($l->current_row['iso']); })
    ->setModel(new Country($db))
    ->setLimit(20);
```

screen shot 2017-12-21 at 04 18 17

 - Documentation: [docs/lister.rst](docs/lister.rst)
 - Demo: [demos/lister.php](demos/lister.php)

### CODE REVIEW

File: src/Lister.php [hunk 1]
```diff
@@ -33,15 +37,31 @@ public function renderView()
...
                 continue;
             }
 
-            $rowHTML = $this->t_row->set($this->current_row)->render();
-            $this->template->appendHTML('rows', $rowHTML);
+            $this->t_row->trySet('_title', $this->model[$this->model->title_field]);
+            $this->t_row->trySet('_href', $this->app->url(['id'=>$this->current_id]));
+            $this->t_row->trySet('_id', $this->current_id);
+
+            if ($this->t_row == $this->template) {
```
Comment (DarkSide666) [inline src/Lister.php]: I slightly don't like this condition to be executed on every loop. That can have bad performancevespeciallyif templates are longer. What if we normalize template in `initChunks()` - if template don't have `{rows}{row}` tags, then add them and further work with only them in `renderView()`?

File: src/View.php [hunk 2]
```diff
@@ -197,12 +197,15 @@ public function setModel(\atk4\data\Model $m)
...
+     * @param array $data   Array of data
+     * @param array $fields Limit model to particular fields
      *
      * @return \atk4\data\Model
      */
-    public function setSource(array $data)
+    public function setSource(array $data, $fields = null)
     {
+        $this->setModel(new \atk4\data\Model(new \atk4\data\Persistence_Static($data)), $fields);
+        /*
```
Comment (DarkSide666) [inline src/View.php]: Should we leave all that uncommented block here?

File: src/Lister.php (hunk 1 above)
Comment (romaninsh) [reply src/Lister.php]: i actually had some problems with it. Don't, mind if that's in a separate PR. Performance wise, it's just comparison, shouldn't be too bad.

File: src/View.php (hunk 2 above)
Comment (romaninsh) [reply src/View.php]: I'll remove.

**Pull request merged**